### ISSUE TITLE
manage columns: don't display empty fields

### ISSUE THREAD

**Issue by sergiocontrino** [intermine/im-tables#167]
manage columns displays also attributes with no data, not like the query builder default
see wormmine issue
https://github.com/WormBase/website/issues/6795

**Comment by yochannah** [comment]
@sergiocontrino fyi, migrated this ticket to im-tables as it's the repo where the visual coponent is managed. imjs is just the communication library

**Comment by sergiocontrino** [comment]
oops..

**Comment by Dhruvi16** [comment]
I would like to work on this.

**Comment by Emuohwo** [comment]
@sergiocontrino link to issue broken, kindly check it

**Comment by yochannah** [comment]
Ooh, that is broken.

Dear @nuin - any idea what happened to the wormbase issue link?

**Comment by nuin** [comment]
@yochannah  Our issues are private now. I will paste the contents of the ticket

**Comment by nuin** [comment]
This has caused quite a bit of confusion for several of our users (and myself):

The data model shown on the QueryBuilder page "Model browser" shows the attributes for the pertinent class that actually have data loaded. When a user runs a query and then explores the data model for the same class from the "Manage Columns" > "Add a Column" option, several additional attributes appear as options, all of which (so far as I can tell), have no data and are misleading. For example, the Gene class:

https://docs.google.com/presentation/d/13pgKFL-HrXkvw1Lm_Z1Nhm_X6tTXzT0z7KGhsjrWhy4/edit?usp=sharing

Comparing the two data models shown for the Gene class shows that the "Manage Columns" > "Add a Column" option shows 18 data and object properties that aren't available in the QueryBuilder "Model Browser" and all of which have no data associated.

I think it is important that we only show the valid attributes that have data in both views. We should eliminate the attributes in the "Manage Columns" > "Add a Column" view that have no data and are not present in the main model.

**Comment by yochannah** [comment]
thanks @nuin! 

@Dhruvi16 - I've assigned the issue to you, let us know if you have questions.

**Comment by Dhruvi16** [comment]
Hey! I was able to reproduce the issue. But I am a little overwhelmed with the codebase and was not able to find where the list comes from. So can anyon ehelp me with this?
Thanks in advance.

**Comment by yochannah** [comment]
@Dhruvi16 - I think the files to edit are one of these: https://github.com/intermine/im-tables/search?q=columns.FindColumnToAdd&unscoped_q=columns.FindColumnToAdd 

let me know if that helps!

**Comment by Dhruvi16** [comment]
I do not think you got my question. So basically the list of 18 data and object properties are not static obviously. So I am unable to figure out where this list comes from. Could you please help me with that?
Thank you
